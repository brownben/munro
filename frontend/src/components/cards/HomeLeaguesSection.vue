<template>
  <section class="w-full pt-8 pb-10 bg-gray-100 border-t border-main-100">
    <div class="max-w-screen-xl mx-auto sm:px-6 lg:px-8">
      <div class="flex flex-row flex-wrap justify-between px-6 mx-auto mb-4">
        <h2
          class="inline-block py-2 text-lg leading-6 tracking-wide uppercase font-heading text-main-600"
        >
          Leagues
        </h2>
        <router-link
          to="/leagues"
          class="inline-block px-4 pt-2 text-sm leading-6 tracking-wide text-right uppercase transition duration-300 text-main-600 font-heading hover:bg-main-100 focus:bg-main-100 rounded-shape"
        >
          View All Leagues â†’
        </router-link>
      </div>

      <div class="grid grid-cols-2 gap-6 px-6 sm:gap-8 md:grid-cols-3">
        <LeagueOverviewCard
          title="(Not) Sprintelope"
          description=" Socially Distant Sprint/ Urban Orienteering Around Edinburgh"
          website="https://en-gb.facebook.com/sprintelope/"
          alternativeLink="/leagues/(Not) Sprintelope 2020"
          :larger="false"
        />
        <LeagueOverviewCard
          title="Fight with the Night"
          description="Thursday Night Orienteering Events Around Edinburgh"
          website="https://euoc.wordpress.com/"
          alternativeLink="/leagues/Fight with the Night 19-20"
          :larger="false"
        />
        <LeagueOverviewCard
          title="Northern Night Cup"
          description="Night Orienteering Events in the North of Scotland"
          website="https://www.orienteering.scot/northern-night-cup/"
          alternativeLink="/leagues/Northern Night Cup 19-20"
          :larger="false"
        />
      </div>
    </div>
  </section>
</template>
<script>
import LeagueOverviewCard from '@/components/cards/LeagueOverviewCard.vue'

export default {
  components: {
    LeagueOverviewCard,
  },
}
</script>
