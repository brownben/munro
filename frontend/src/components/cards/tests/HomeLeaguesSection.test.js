import { mount } from '@vue/test-utils'
import HomeLeaguesSection from '@/components/cards/HomeLeaguesSection.vue'

test('Is a Vue Instance', () => {
  const wrapper = mount(HomeLeaguesSection, { stubs: ['router-link'] })
  expect(wrapper.isVueInstance()).toBeTruthy()
})

test('Renders Correctly', () => {
  const wrapper = mount(HomeLeaguesSection, { stubs: ['router-link'] })
  expect(wrapper.element).toMatchSnapshot()
})
