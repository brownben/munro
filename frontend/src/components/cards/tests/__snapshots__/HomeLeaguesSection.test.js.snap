// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
<section
  class="w-full pt-8 pb-10 bg-gray-100 border-t border-main-100"
>
  <div
    class="flex flex-row flex-wrap justify-between w-10/12 mx-auto mb-4 xl:w-3/4"
  >
    <h2
      class="inline-block py-2 text-lg leading-6 tracking-wide uppercase font-heading text-main-600"
    >
      
      Leagues
    
    </h2>
     
    <router-link-stub
      class="inline-block px-4 pt-2 text-sm leading-6 tracking-wide text-right uppercase transition duration-300 text-main-600 font-heading hover:bg-main-100 focus:bg-main-100 rounded-shape"
      to="/leagues"
    >
      
      View All Leagues â†’
    
    </router-link-stub>
  </div>
   
  <div
    class="grid w-10/12 grid-cols-2 gap-8 mx-auto xl:w-3/4 md:grid-cols-3"
  >
    <section
      class="flex flex-col items-center justify-around col-span-2 px-4 pt-6 pb-4 text-center bg-white shadow-md md:col-span-1 md:px-3 xl:px-4 rounded-shape-xl"
    >
       
      <h2
        class="mt-2 mb-4 text-2xl font-bold leading-6 xl:text-25xl font-heading text-main-900"
      >
        
    (Not) Sprintelope
  
      </h2>
       
      <p
        class="mb-2 leading-5 text-opacity-75 text-main-900 font-heading"
      >
        
     Socially Distant Sprint/ Urban Orienteering Around Edinburgh
  
      </p>
       
      <router-link-stub
        class="mt-3 button"
        to="/leagues/(Not) Sprintelope 2020"
      >
        View League
      </router-link-stub>
    </section>
     
    <section
      class="flex flex-col items-center justify-around col-span-2 px-4 pt-6 pb-4 text-center bg-white shadow-md md:col-span-1 md:px-3 xl:px-4 rounded-shape-xl"
    >
       
      <h2
        class="mt-2 mb-4 text-2xl font-bold leading-6 xl:text-25xl font-heading text-main-900"
      >
        
    Fight with the Night
  
      </h2>
       
      <p
        class="mb-2 leading-5 text-opacity-75 text-main-900 font-heading"
      >
        
    Thursday Night Orienteering Events Around Edinburgh
  
      </p>
       
      <router-link-stub
        class="mt-3 button"
        to="/leagues/Fight with the Night 19-20"
      >
        View League
      </router-link-stub>
    </section>
     
    <section
      class="flex flex-col items-center justify-around col-span-2 px-4 pt-6 pb-4 text-center bg-white shadow-md md:col-span-1 md:px-3 xl:px-4 rounded-shape-xl"
    >
       
      <h2
        class="mt-2 mb-4 text-2xl font-bold leading-6 xl:text-25xl font-heading text-main-900"
      >
        
    Northern Night Cup
  
      </h2>
       
      <p
        class="mb-2 leading-5 text-opacity-75 text-main-900 font-heading"
      >
        
    Night Orienteering Events in the North of Scotland
  
      </p>
       
      <router-link-stub
        class="mt-3 button"
        to="/leagues/Northern Night Cup 19-20"
      >
        View League
      </router-link-stub>
    </section>
  </div>
</section>
`;
